streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1459,6 +1467,43 @@ private void verifyEOSTransactionTimeoutCompatibility() {
         }
     }
 
+    private void verifyClientTelemetryConfigs() {
+        final String mainConsumerMetricsPushKey = mainConsumerPrefix(ENABLE_METRICS_PUSH_CONFIG);
+        final String adminClientMetricsPushKey = adminClientPrefix(ENABLE_METRICS_PUSH_CONFIG);
+        final boolean streamTelemetryEnabled =
+                !originals().containsKey(ENABLE_METRICS_PUSH_CONFIG) || (boolean) Objects.requireNonNull(
+                        parseType(ENABLE_METRICS_PUSH_CONFIG, originals().get(ENABLE_METRICS_PUSH_CONFIG), Type.BOOLEAN),
+                        "Can't parse " + ENABLE_METRICS_PUSH_CONFIG + " because it's null");
+
+        final boolean mainConsumerMetricsDisabled =
+                !originals().containsKey(mainConsumerMetricsPushKey) || (boolean) Objects.requireNonNull(
+                parseType(mainConsumerMetricsPushKey, originals().get(mainConsumerMetricsPushKey), Type.BOOLEAN),
+                "Can't parse " + mainConsumerMetricsPushKey + " because it's null");
+
+        final boolean adminMetricsEnabled =
+                !originals().containsKey(adminClientMetricsPushKey) || (boolean) Objects.requireNonNull(
+                        parseType(adminClientMetricsPushKey, originals().get(adminClientMetricsPushKey), Type.BOOLEAN),
+                        "Can't parse " + adminClientMetricsPushKey + " because it's null");
+
+        final String baseMetricsMisconfigurationMessage = "KafkaStreams has metrics push enabled" +
+                " but the %s metrics push is disabled. Enable " +
+                " metrics push for the %s";
+
+        if (streamTelemetryEnabled) {
+            if (!mainConsumerMetricsDisabled && !adminMetricsEnabled) {
```

Comment:
Given that you use `!...Disabled` and `!...Enabled` would it make sense to flip the variables from "disabled -> enable" and "enable -> disabled" to avoid the negation here (and below)?

Also, I think we should throw if `streamTelemetryEnabled && mainConsumerMetricsDisabled` not if "main consumer enable" -- it seem the code works correctly though, as we compute `mainConsumerMetricsDisabled` incorrectly above, and thus there is two error which cancel out each other?